compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -264,30 +178,66 @@ private void validateRecordFieldsAnnotation(SyntaxNodeAnalysisContext ctx, Node
                     validateReadOnly(ctx, fieldNode.readonlyKeyword().isPresent(), fieldNode.location());
                 }
                 Optional<MetadataNode> metadataNode = fieldNode.metadata();
-                metadataNode.ifPresent(node -> validateAnnotation(ctx, node, fieldNode.typeName().toSourceCode().trim(),
+                metadataNode.ifPresent(node -> validateAnnotationFields(ctx, node,
+                        fieldNode.typeName().toSourceCode().trim(),
                         fieldNode.location(), memberNodes, fieldNode.fieldName().text().trim()));
             } else if (field instanceof RecordFieldWithDefaultValueNode) {
                 RecordFieldWithDefaultValueNode fieldNode = (RecordFieldWithDefaultValueNode) field;
                 if (primaryKeys.contains(fieldNode.fieldName().text().trim())) {
                     validateReadOnly(ctx, fieldNode.readonlyKeyword().isPresent(), fieldNode.location());
                 }
                 Optional<MetadataNode> metadataNode = fieldNode.metadata();
-                metadataNode.ifPresent(node -> validateAnnotation(ctx, node, fieldNode.typeName().toSourceCode().trim(),
+                metadataNode.ifPresent(node -> validateAnnotationFields(ctx, node,
+                        fieldNode.typeName().toSourceCode().trim(),
                         fieldNode.location(), memberNodes, fieldNode.fieldName().text().trim()));
             }
         }
     }
 
+    private void checkFieldAnnotationPresent(SyntaxNodeAnalysisContext ctx, Node recordNode) {
+        RecordTypeDescriptorNode recordTypeDescriptorNode = (RecordTypeDescriptorNode) recordNode;
+        NodeList<Node> fields = recordTypeDescriptorNode.fields();
+        for (Node field : fields) {
+            if (field instanceof RecordFieldNode) {
+                RecordFieldNode fieldNode = (RecordFieldNode) field;
+                Optional<MetadataNode> metadataNode = fieldNode.metadata();
+                metadataNode.ifPresent(node -> validateAnnotation(ctx, node, fieldNode.location()));
+            } else if (field instanceof RecordFieldWithDefaultValueNode) {
+                RecordFieldWithDefaultValueNode fieldNode = (RecordFieldWithDefaultValueNode) field;
+                Optional<MetadataNode> metadataNode = fieldNode.metadata();
+                metadataNode.ifPresent(node -> validateAnnotation(ctx, node, fieldNode.location()));
+            }
+        }
+    }
+
+    private void validateAnnotation(SyntaxNodeAnalysisContext ctx, MetadataNode metadataNode, Location location) {
+        NodeList<AnnotationNode> annotations = metadataNode.annotations();
+        for (AnnotationNode annotation : annotations) {
+            String annotationName = annotation.annotReference().toSourceCode().trim();
+            if (annotationName.equals(Constants.AUTO_INCREMENT)) {
+                if (!hasPersistAnnotation) {
```

Comment:
Why is this check repeated? Isn't it done at the first instance itself

Reply:
Fixed in #73